### ISSUE TITLE
Slow computation when no gradients exist w.r.t a layer

### ISSUE THREAD

**Issue by sayakpaul** [tensorflow/tensorflow#40432]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Colab
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): 
- TensorFlow version (use command below): 2.2.0
- Python version:
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:
- GPU model and memory:

**Describe the current behavior**

I am trying to train a barebone Neural Machine Translation model with Luong attention (`dot` for calculating the alignment scores). I have gone for a custom training loop and I am getting the following warning which does make sense to me since gradients are not calculated for Embedding layers as far as I know:

![image](https://user-images.githubusercontent.com/22957388/84569556-35ad9980-ada5-11ea-9540-68137de72532.png)

It also slows the computation and one immediate workaround I could imagine was to wrap the single training step within a function and decorate with `tf.function`. But I really could not figure a way out. 

**Describe the expected behavior**

A workaround to enhance the computation or the overall training loop in general. 

**Standalone code to reproduce the issue**
[Colab Notebook](https://colab.research.google.com/drive/1GpMiXVonmQKQdiOTcDW9XgDnp9K0iL0b?usp=sharing)

**Comment by ravikyram** [comment]
I have tried in colab with TF version 2.2 and was able to reproduce the issue.Please, find the gist [here](https://colab.research.google.com/gist/ravikyram/38cf90d8d3cbe7257fc9843047922038/untitled21.ipynb).Thanks!

**Comment by sayakpaul** [comment]
Hi. Any updates on this?
